pytorch_lightning/metrics/bleu.py

```diff
@@ -0,0 +1,143 @@
+from collections import Counter
+import math
+import re
+
+
+def math_log(num: int):
+    return -99999999 if num == 0.0 else math.log(num)
+
+
+def best_match_len(translate_list: list, reference_list: list):
+    """Finding the closet length of the reference texts with respect to translated text
+
+    Args:
+        translate_list: A list of translated text
+        reference_list: A list of reference texts
+
+    Return:
+        ref_len: the best match length of text of the reference lists with translated text
+        translate_len: the length of translated text
+    """
+    translate_len = len(translate_list)
```

Comment:
this also needs to be a pure pytorch implementation.

Reply:
so the inputs to the `bleu_score` should all be torch.Tensor?
Not string?